File: passenger-service/src/main/java/com/software/modsen/passengerservice/entity/PassengerRating.java
package com.software.modsen.passengerservice.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "passenger_ratings")
@Builder
@ToString
public class PassengerRating {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "passenger_rating_id")
    private Long id;

    @Column(name = "rating")
    private double rating;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "passenger_id")
    private Passenger passenger;
}

File: passenger-service/src/main/kotlin/com/software/modsen/passengerservice/entity/PassengerRating.kt
package com.software.modsen.passengerservice.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne
import jakarta.persistence.Table
import jakarta.persistence.Version

@Entity
@Table(name = "passenger_ratings")
class PassengerRating {
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    @field:Column(name= "passenger_rating_id")
    var id: Long?

    @field:Column(name = "rating")
    var rating: Double

    @field:OneToOne(fetch = FetchType.EAGER)
    @field:JoinColumn(name = "passenger_id")
    var passenger: Passenger?

    @field:Column(name = "version")
    @field:Version
    private val version: Long?

    constructor() {
        id = null
        rating = 0.0
        passenger = null
        version = null
    }

    constructor(rating: Double, passenger: Passenger?) : this() {
        this.rating = rating
        this.passenger = passenger
    }

    constructor(id: Long?, rating: Double, passenger: Passenger?) : this() {
        this.id = id
        this.rating = rating
        this.passenger = passenger
    }
}